### DISCUSSION TITLE
DIY Hardware possible?

### DISCUSSION THREAD

**Discussion by jankosso** [john30/ebusd#598] [Q&A]
I would like to order an adaptor, but i need one soon for my first experiments. My idea is to copy bus driver circuit including PIC on a Breadboard and replace isolators with available ones and add a NodeMCU.

Is it possible to flash the PIC Hexfile from the project page directly with a Microchip tool?

**Comment by john30** [comment]
yes

**Comment by koen-lee** [comment]
As an alternative, you can perhaps experiment with a read-only adapter, as found on https://ebus-wiki.org/doku.php/ebus/ebuskonverter
or adapt the simpler isolated ebus stage of https://gitlab.com/fromeijn/ebuzzz-adapter/-/blob/master/ebuzzz-adapter.pdf to the PIC, as it does not need expensive high speed isolators with DC/DC converters but optocouplers.
That might be an idea for any v4 designers too 😄

**Reply by john30** [reply]
that's not a good idea as the power consumption from the bus is too high. that's why we have started with the dc/dc couplers in adapter v2

**Reply by koen-lee** [reply]
I had in mind to have the PIC on the external-powered device side, just the signal conditioning on the bus-powered isolated side.
Basically the circuit from the spec page 25:
https://ebus-wiki.org/lib/exe/fetch.php/ebus/spec_prot_12_v1_3_1_e.pdf which is listed as 1.3 mA allowing 53 bus participants.
The VRC720 bus-powered thermostat draws more.
I'm sure that this has been considered, so I am curious about the reasoning to do it the hard and expensive way.
What is the best place to discuss hardware?

**Reply by john30** [reply]
when drawing too much current from the bus, the signal converts from rectangle to a rather sawtooth like one and this makes communication harder with every additional participant on the line. the devices from the heating manufacturers usually have a super capacitor on board to compensate high power to be drawn from the bus in idle mode only

**Reply by koen-lee** [reply]
Thank you for your reply.
So the linked specifications are just wrong, and there is no better place to discuss hardware?

I understand that drawing power from the bus has consequences (10.3.1, pg 22), but the specs specifically say power draw is intended use (10.3, pg 21).
Lower power opto's than the SFH6136 from the example circuit can easily be found.

What is, in your practical experience, the maximum permissible current draw?

**Reply by john30** [reply]
the specs are right on a theoretical base, but as so often practice prooves different. IIRC it has to be below 1mA to not have bad influence on the signal. it depends a lot on the wiring esp. cable lengths, the manufacturer, the number of devices connected and so on. you can't just tell from the specs.

what's the purpose of the exhausting discussion about a better place for discussion?? just discuss it elsewhere then.

**Reply by m0deller** [reply]
I tried powering [this circuit](https://oshwlab.com/peterpeter/nodemcu-ebus-controller)and the Wemos mini via ebus power - vaillant aerotherm+ . The ebus stopped working .. to much load.

**Reply by B-G-T** [reply]
The use of dc/dc couplers from V2 onwards was for sure the right decision.
I just wonder why it's better to have the PIC in the domain supplied via the dc/dc converter and not supplied directly?

**Comment by m0deller** [comment]
I've built [this circuit](https://oshwlab.com/peterpeter/nodemcu-ebus-controller) on a breadboard and used the Adapter3 firmware on a Wemos mini. Works well. Been running for months.

**Reply by dk7jb** [reply]
Hello m0deller
How do you read the Adapter3 firmware? Do you maybe work with Home Assistant?

**Reply by m0deller** [reply]
the Adapter3 firmware is hosted here in GitHub. 
If you mean by 'read', how do I access the ebus data, then I use MQTT , node-red and the open energy monitor (emon).
